text/0000-release-process.md

```diff
@@ -0,0 +1,194 @@
+- [code freeze](#code-freeze) - branch of code into a [release branch](#release-branches) for on going [UAT](#user-acceptance-testing)
+- [release execution](#release-execution) - the step of finalizing and publishing a release
+
+```text
+ release X
+
+ +---------------+      +-------------+      +---------+
+ |  development  | ---> | code freeze | ---> | release |
+ +---------------+      +-------------+      +---------+
+
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Timetable
+
+_The following schedules are suggestive and subject to change based on practicality._
+
+* Release Cadence: **Monthly** on the **1st Tuesday** of the month 
+* Code Freeze: **5 days** prior to scheduled release
+* CCB Review: **daily**
+
+### Concurrency
+
+Due to the nature of [release branches](#release-branches), this enables development to continue on future releases
+while finalizing the current release. 
+
+```text
+ release X
+
+ +---------------+      +-------------+      +---------+
+ |  development  | ---> | code freeze | ---> | release |
+ +---------------+      +-------------+      +---------+
+
++---------------------------------------------------------------------------------+
+
+                        release Y
+
+                        +---------------+      +-------------+      +---------+
+                        |  development  | ---> | code freeze | ---> | release |
+                        +---------------+      +-------------+      +---------+
+```
+
+### Terminology
+
+#### User Acceptance Testing
+
+User acceptance testing (UAT) is the process of manually verifying that new features and fixes are **complete** and 
+**don't introduce issues**. Typically, there are two strategies:
+
+1. Rolling UAT - User acceptance is done per issue as the issue is completed.
+2. Feature-Complete UAT - User acceptance is done when all features are complete.
+
+More commonly Feature-Complete User Acceptance is used in practice because any additional changes could have unexpected
+side-effects invalidating prior completed UATs. In our case, Feature-Complete UAT fit perfectly during the
+[code freeze](#code-freeze) phase.
+
+#### Code Freeze
+
+A code freeze for a particular release is a period of time when changes are analyzed and vetted before being added to
+a release.
+
+The following take place:
+
+1. At the beginning, a release branch is created (`release/<version>`).
+2. At the beginning, a [change control board](#change-control-board) is organized on a daily basis to review additional
+requested changes.
+3. During, only changes approved by the CCB are merged into the release branch.
+
+#### Change Control Board
+
+A change control board would consist of core developers that, on a daily basis, organize changes that might
+be suitable for [release branches](#release-branches) during the code freeze.
+
+The board members typically decide on whether a change should be included on the release based on the following criteria:
```

Comment:
We should probably formalize how one joins the CCB or who comprises the CCB initially. Maybe in another RFC?

Reply:
Good question. Given this is a community project rather than a team I could see some strategies that differ from common practices. 

FWIW, typically it would involve product owner, development lead, and QA lead.

Reply:
Is the CCB for now not just the maintainers for the subteam the project falls under?

Reply:
Yes, I think we agreed on that. I've updated the RFC to reflect that.